tests/tools/test_check_test_tethering.py

```
@@ -0,0 +1,589 @@
+    [
+        # A -k / -m filter narrows which tests run but does not un-run the file.
+        ("pytest -v -s kernels -k 'not slow'", "kernels/test_a.py", True),
+        ("pytest -v -s kernels -m core_model", "kernels/test_a.py", True),
+        # A value-taking option's value is never mistaken for a path.
+        ("pytest -m core_model", "core_model", False),
+        ("pytest -k kernels", "kernels/test_a.py", False),
+        ("pytest -n 4 kernels/test_a.py", "kernels/test_a.py", True),
+        # Buildkite shard flags carry a $$VAR and must not swallow the path.
+        (
+            "pytest -v -s kernels --shard-id=$$BUILDKITE_PARALLEL_JOB "
+            "--num-shards=$$BUILDKITE_PARALLEL_JOB_COUNT",
```

Ruff (0.16.3)

[warning] Unparenthesized implicit string concatenation in collection

Did you forget a comma?

(ISC004)